ctask/doer.go

```diff
@@ -0,0 +1,68 @@
+package ctask
+
+import (
+	"context"
+	"runtime"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type DoOpt func(cfg *DoConfig)
+type DoConfig struct {
+	WorkerNum int
+}
+
+// Do execute tasks using the given executor function,
+// and return the results in the same order as the given tasks respectively.
+// It stops executing remaining tasks after any first error is encountered.
```

Comment:
as discussed in private chat, [g.Go ](https://github.com/trustwallet/go-libs/pull/197/files#diff-0e3e913f3d30c66cc9aa10ed0ae5c1ab18e5c966cadaa859843ed541a2bbd74bR36) will cancel the context when any operation returns an error. So when an error is encountered, the rest will not be executed anymore. 

Technically we still spawn goroutines, but it will exit immediately